docs/src/how-to/4-develop.md

```diff
@@ -0,0 +1,136 @@
+
+Let's select the environment of our KomaMRI project:
+
+![](../assets/dev-vscode-select-komamri-environment.png)
+
+Now, open a Julia REPL:
+
+![](../assets/dev-open-julia-repl.png)
+
+In the Julia REPL you have to create the `Manifest.toml` file with the `develop` command considering de local folders `KomaMRIBase`, `KomaMRICore`, `KomaMRIFiles` and `KomaMRIPlots`. Additionally you can install all the dependencies listed in the `Manifest.toml` with the `instantiate` command:
+```julia-repl
+(KomaMRI) pkg> dev ./KomaMRIBase ./KomaMRICore ./KomaMRIFiles ./KomaMRIPlots
+
+(KomaMRI) pkg> instantiate
+```
+
+With these steps your environment is setup. You can now run the KomaMRI user interface:
+```julia-repl
+julia> using KomaMRI
+
+julia> KomaUI()
+```
+![](../assets/dev-launch-ui.png)
+
+Let's see what is the default phantom in the user interface by clicking in the `Phantom dropdown` and then press the `View Phantom` button:
+
+![](../assets/dev-default-phantom-brain.png) 
+
+You can close the KomaMRI user interface.
+
+
+## Make changes to remote repos
+
+So far so good. However you want to make some changes to your Github Fork and even you want to contribute to the KomaMRI project. But first, let's make some local changes in the local git repo. In the following example we will change the default phantom displayed in the user interface.
+
+First create a branch with a descriptive name, in this case `patch-ui-default-phantom`:
+
+![](../assets/dev-branch.png)
+![](../assets/dev-branch-name.png)
+
+In this new branch, we are going to edit the `setup_phantom` function located in the file `src/ui/ExportUIFunctions.jl`. We simply replace the line where a brain phantom is defined by the function `pelvis_brain2D`: 
+
+![](../assets/dev-branch-edit.png)
+
+Then, you need to compile the function `setup_phantom`, simply put the cursor in a place inside the function `setup_phantom` and then press `Alt + Enter`:
+
+![](../assets/dev-branch-compile.png)
+
+Now, let's see what is the new default phantom in the user interface by clicking in the `Phantom dropdown` and then press the `View Phantom` button:
+
+```julia-repl
+julia> KomaUI()
+```
+
+![](../assets/dev-default-phantom-pelvis.png)
+
+Everything looks good, let's stage the changes:
+
+![](../assets/dev-stage-changes.png)
+
+And commit the changes:
+
+![](../assets/dev-commit-changes.png)
+
+Let's upload these changes in your github fork or to the official KomaMRI repo by selecting the `origin` or the `upstream` option:
+
+![](../assets/dev-publish.png)
+
+![](../assets/dev-pullreq.png)
+
+![](../assets/dev-pullreq-name.png)
+
+
+## About Julia Environments and Subdirectories
+
+So far, the `KomaMRI` package have 4 subdirectories `KomaMRIBase`, `KomaMRICore`, `KomaMRIPlots` and `KomaMRIFiles`. These subdirectories are proper julia packages by themselves, so they have their own `Project.toml` which need to resolve an environment defined ultimately by a `Manifest.toml`.
```

Comment:
So far, the `KomaMRI` package has 4 subdirectories: `KomaMRIBase`, `KomaMRICore`, `KomaMRIPlots` and `KomaMRIFiles`. These subdirectories are julia packages by themselves, so they have their own `Project.toml` which is needed to resolve the environment ultimately defined by `Manifest.toml`. The directories are not separate git submodules.